keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -107,123 +116,81 @@ serves the resourceVersion="0" list requests from reflectors today.
-
 Consistent GET requests will continue to be served directly from etcd. We will
 only serve consistent LIST requests from cache.
 
-Important: We are planning to set the progress notify interval to 250ms, which will introduce up to 250ms latency to consistent LIST requests.
-
-Optional: For some (but not all) of the etcd progress watch events, also create a
-kubernetes "bookmark" watch event and send it to kube-apiserver clients so that
-reflectors and shared informers are kept up-to-date. The benefit of this is that
-it minimizes the chance that these clients will end up with an out-of-date
-resource version and need to relist (which can impact scalability). See [Watch
-Bookmarks](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190206-watch-bookmark.md)
-for details.
+As a consequence of this number of etcd progress notification will increase
+number. [Watch Bookmarks KEP] utilized etcd periodic progress notifications to
+notify kube-apiserver clients of their progress. It used periodic watch
+notifications that are naturally limited on etcd side. To maintain the same
+behavior we will limit watch bookmarks to send them only one per 5 seconds.
 
-#### Determining if etcd is sending progress notify events
-
-It is possible to automatically determine if etcd is sending progress notify events.
-Each watch cache could keep track of when it received the last progress notify event.
-If it has been sufficiently long since the last one was received, or if none have
-ever been received, the watch cache should assume etcd is not sending progress notify,
-and not attempt to serve consistent reads from cache, falling back to serving
-the reads directly from etcd.
+[getCurrentResourceVersionFromStorage]: https://github.com/kubernetes/kubernetes/blob/3f247e59edfd4083242ad7271d076a38291760ff/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L1246-L1278
+[Watch-List KEP]: /keps/sig-api-machinery/3157-watch-list
+[Watch Bookmarks KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/956-watch-bookmark
 
 ### Risks and Mitigations
 
-Configuring etcd to send progress notify events may have performance implications.
-@mm4tt and @wojtek-t have run [some
-experiments](https://github.com/kubernetes/kubernetes/pull/86769) that suggest
-that a progress notify interval of 50ms results in a noticeable increase in both
-etcd and kube-apiserver CPU utilization and some increased kube-apiserver serving
-latency. We intend to address this by first trying a 250ms progress notify interval,
-if ths performs well we will use that as our interval. If not, we will dig in
-to figure out why the events are impacting performance and see if we can optimize
-it away.
+### Dependency on manual watch progress notifications
 
-## Design Details
+<<[UNRESOLVED @serathius]>>
+Validate performance implications of using `WatchProgressRequest`.
```

Comment:
Please add an additional context that I asked about above

```
Progress is requested on client, so all watches opened with this client will get the notification.
```

Reply:
And add a consideration that currently we have etcd-client per resource, which makes that fine,
but any attempt to reuse etcd-client across type
xref https://github.com/kubernetes/kubernetes/pull/114458 @enj 
would make that problematic